lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -361,21 +361,18 @@ public function getCreatePrimaryKeySQL(Index $index, $table)
     protected function getCreateColumnCommentSQL($tableName, $columnName, $comment)
     {
         if (strpos($tableName, '.') !== false) {
-            [$schemaSQL, $tableSQL] = explode('.', $tableName);
-            $schemaSQL              = $this->quoteStringLiteral($schemaSQL);
-            $tableSQL               = $this->quoteStringLiteral($tableSQL);
+            [$schemaName, $tableName] = explode('.', $tableName, 2);
```

Comment:
That looks related to the fix: we don't quote the identifier because it should not be passed as a string literal

Reply:
yes

Reply:
`2` assumes that the schema name cannot contain dots but the table can. Why?

Reply:
see https://3v4l.org/UrgeT - this is only improvement of the original `explode`, which can return more than 2 result (and the 3rd and later result was silently discarded previously)

Reply:
The question still is: why do you think the previously discarded part after the dot should be attributed to the table name?

Reply:
Because the data of "the previously discarded part after the dot" come from `$tableName` :)